### ISSUE TITLE
Add example for "app subject command -arg" CLI

### ISSUE THREAD

**Issue by philippgille** [natemcmaster/CommandLineUtils#23]
I would like to build a CLI where commands are related to subjects, like the Docker CLI:
- `docker container ls -a`
- `docker image ls -a`
- `docker container run --name some-redis -d redis`

Where `docker` is the application, `container` / `image` etc. the subject and `ls` / `run` etc. the command, with commands belonging to the subject and with arguments only applying to the specific command.

The [subcommand example](https://github.com/natemcmaster/CommandLineUtils/blob/912626722541542ba8063797958344ae15f9c45b/samples/Subcommands/Program.cs) only used `app command -arg`, so the level of "subjects" is missing. But it seems to be possible with building a hierarchy with the `Parent` attribute, at least from how I understand the [subcommand tests](https://github.com/natemcmaster/CommandLineUtils/blob/912626722541542ba8063797958344ae15f9c45b/test/CommandLineUtils.Tests/SubCommandTests.cs).

It would be nice to have some kind of documentation about how to "nest" subcommands to create a Docker-like CLI.

And a question: In a Docker-like CLI it shouldn't be possible to use multiple subjects in one line, e.g. `docker container ls -a image ls -a`. But for example [this line](https://github.com/natemcmaster/CommandLineUtils/blob/912626722541542ba8063797958344ae15f9c45b/test/CommandLineUtils.Tests/SubCommandTests.cs#L120) in the tests looks like that's currently possible when using subcommands in CommandLineUtils. Is there a way to restrict that?

**Comment by natemcmaster** [comment]
Check out the sample added in #25. 

> In a Docker-like CLI it shouldn't be possible to use multiple subjects in one line, e.g. docker container ls -a image ls -a.

Luckily, this is not the behavior of subcommands. The behavior you're seeing here was a test to ensure that in a multi-level subcommand app, you can have command-line flags on top, middle, and bottom levels. For example, if `docker container` had a flag `--magic`, this library would support `docker container --magic ls -a image`

**Comment by philippgille** [comment]
Ah okay, that's actually a nice feature.

I'll probably figure it out when trying it out by myself, but maybe you can already give me a heads-up: In https://github.com/natemcmaster/CommandLineUtils/pull/25 the `git` example uses parents, the `docker` example uses nested classes. Parents are for utilizing shared functionality in subclasses, right? And is there a way to have a docker-like CLI without nested classes? Looks like the type is just needed in for example `Subcommand("ls", typeof(List))` and `List` could be an `internal` class, defined in `List.cs`, right?

**Comment by natemcmaster** [comment]
Nested classes and inheritance are two things you can do, but you don't have to. You could structure an app with simple classes like this:
```c#
[Subcommand("container", typeof(Container)]
class Docker
{ }

[Subcommand("ls", typeof(ListContainer)]
class Container
{ } 

class ListContainer
{
  [Option("-a|--all")]
  public bool All { get; }
} 
```

**Comment by natemcmaster** [comment]
Closed the issue by adding #25, but feel free to continue discussion/questions about subcommands.
